src/keep_gpu/utilities/gpu_info.py

```
@@ -158,16 +162,22 @@ def _query_rocm() -> List[Dict[str, Any]]:
+                idx,
+                monitor_count=monitor_count,
+            )
             util = None
-            try:
-                util = int(rocm_smi.rsmi_dev_busy_percent_get(idx))
-            except Exception as exc:
-                logger.debug("ROCm util query failed for %s: %s", idx, exc)
+            if physical_id is not None:
+                try:
+                    util = int(rocm_smi.rsmi_dev_busy_percent_get(physical_id))
+                except Exception as exc:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)